### ISSUE TITLE
Step-by-step debugger

### ISSUE THREAD

**Issue by djmattyg007** [spacechem-community-developers/SpaceChem-Community-Edition#70]
The ability to move a solution forward by exactly one cycle at a time would be amazing for debugging complicated setups.

**Comment by csaboka** [comment]
I agree that this would be very handy, and it doesn't seem to be hard to add it to the simulation engine. The tricky part is, again, the UI. My best idea so far is to handle pressing the pause button when the simulation is already paused by letting it run to the beginning of the next cycle, but that isn't discoverable at all.

I'm open to any ideas about an intuitive UI for this.

**Comment by whoever314** [comment]
Could you show a '1' on the pause button when the sim is paused or not running, and the usual pause symbol when it's running?

**Comment by csaboka** [comment]
This has been implemented in version 6.5, released just now. I ended up using the Pause button idea, with a hacky button saying ">|" to indicate that it will play and then pause again.

Feel free to open follow-up issues if you find anything that doesn't seem to work right!
